src/lighteval/models/transformers/transformers_model.py

```diff
@@ -508,7 +509,119 @@ def greedy_until_multi_turn(  # noqa: C901
+            # Would we rather truncate the prompt to allow generation to go to max_new_tokens, at the risk
+            # of losing some meaning, or have some generations that are exceedingly short?
+            # The choice we go for here is to avoid truncating the prompt if we can, since it
+            # should have been managed by the prompt creator/few shot manager if requested by the user.
+            inputs = tokenized["input_ids"]
+            context_size = len(inputs[0])
+
+            # left truncate the inputs to the maximum length
+            if max_new_tokens is not None:
+                if context_size + max_new_tokens > self.max_length:
+                    logger.warning(
+                        f"{context_size + max_new_tokens=} which is greater than {self.max_length=}. Truncating context to {self.max_length - max_new_tokens} tokens."
+                    )
+                    context_size = self.max_length - max_new_tokens
+                    if context_size < 0:
+                        logger.critical(
+                            f"{context_size=} is less than 0, either reduce the max_new_tokens or increase model max length."
+                        )
+                        raise ValueError("Context size is less than 0.")
+                    inputs = [input[-context_size:] for input in inputs]
+            else:
+                if context_size > self.max_length:
+                    logger.warning(
+                        f"{context_size=} which is greater than {self.max_length=}. Truncating context to {self.max_length} tokens."
+                    )
+                    context_size = self.max_length
+                    inputs = [input[-context_size:] for input in inputs]
+
+            _outputs = self._generate(
+                inputs=inputs,
+                max_new_tokens=max_new_tokens,
+                stop_tokens=stop_tokens,
+                returns_logits=returns_logits,
+                num_samples=num_samples,
+            )
+
+        for _output in _outputs:
+            output_token_ids = []
+            logprobs_raw = []
+            result = []
+
+            # for output in _output.outputs:
+            output_token_ids.append(_output["output_ids"])
+            # logprobs_raw.append(output.logprobs)
+            result.append(self.tokenizer.decode(_output["output_ids"]))
+
+            if logprobs_raw and output_token_ids and False:
+                logprobs = [
+                    logprobs_raw[0][token_id].logprob
+                    for token_id in output_token_ids[0]
+                ]
+            else:
+                logprobs = []
+
+            input_token_ids = _output["prompt_token_ids"]
+
+            cur_response = GenerativeResponse(
+                result=result,
+                logits=logprobs,
+                generated_tokens=output_token_ids,
+                input_tokens=input_token_ids,
+            )
+            results.append(cur_response)
```

Comment:
I suspect this is indented one level too low (you're going to skip all _outputs except the last one, no?)